### DISCUSSION TITLE
I wanna rewrite pynput_service.py with Rust, I have some questions.

### DISCUSSION THREAD

**Discussion by chiehw** [rustdesk/rustdesk#700] [General]
1. Why are you using pynput instead of [xdotool](https://github.com/rustdesk/rustdesk/blob/cb781fe08c29f41d5b0fb916ab3b401778e1c3dc/libs/enigo/src/linux.rs#L42)? is there something wrong with xdotool? 
2. I plan to rewrite the keyboard part of pynput with [xlib-x11rb](https://github.com/psychon/x11rb) and [evdev](https://crates.io/crates/evdev) (to support wayland), similar to what the pynput library does. Any suggestions?

I've tested that evdev works fine on wayland(requires sudo), x11rb looks mature. @rustdesk

**Comment by rustdesk** [comment]
I use pynput because xdotool causes freeze if you enable multiple languages in Ubuntu.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/292#issuecomment-1059208399
